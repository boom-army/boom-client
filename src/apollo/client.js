import { ApolloClient, InMemoryCache } from "@apollo/client";
import { offsetLimitPagination } from "@apollo/client/utilities";

export const client = new ApolloClient({
  cache: new InMemoryCache({
    typePolicies: {
      Query: {
        fields: {
          feed: offsetLimitPagination(),
          users: offsetLimitPagination(),
        },
      },
    },
  }),
  uri: process.env.REACT_APP_APOLLO_API || "http://locahost:7777",
  headers: {
    authorization: localStorage.getItem("token") || "",
  },
  context: async ({ req }) => {
    const token = req?.headers?.authorization || null;
    const user= token ? JSON.parse(localStorage.getItem("user")) : null;
    return { user, isLoggedIn: !!token };
  },
});
