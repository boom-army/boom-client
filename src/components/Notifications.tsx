import { CustomResponse } from "./CustomResponse";
import { Loader } from "./Loader";
import { useEffect } from "react";
import { useMentionsQuery } from "../generated/graphql";
import { Box, Grid } from "@mui/material";
import { Notification } from "./Notification/Notification";
import InfiniteScroll from "react-infinite-scroll-component";

export const Notifications = ({ refetchProfile }: any) => {
  const { loading, data, fetchMore } = useMentionsQuery({
    variables: {
      offset: 0,
      limit: 10,
    },
  });

  useEffect(() => {
    refetchProfile && refetchProfile();
  }, [data, refetchProfile]);

  const fetchData = () => {
    fetchMore({
      variables: {
        offset: data?.mentions?.length ?? 0,
      },
    });
  };

  if (loading) return <Loader />;
  return (
    <Grid item xs={12}>
      {data?.mentions?.length ? (
        <InfiniteScroll
          dataLength={data?.mentions?.length}
          next={fetchData}
          hasMore={true}
          scrollableTarget="scrollBox"
          loader={
            loading && (
              <Box sx={{ marginTop: "1rem" }}>
                <Loader />
              </Box>
            )
          }
        >
          {data.mentions.length
            ? data.mentions.map((mention: any) => (
                <Notification key={mention.id} mention={mention} />
              ))
            : null}
        </InfiniteScroll>
      ) : (
        <CustomResponse text="Follow some people to get some feed updates" />
      )}
    </Grid>
  );
};
