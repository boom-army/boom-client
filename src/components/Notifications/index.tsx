import { CustomResponse } from "../CustomResponse";
import { Loader } from "../Loader";
import { useEffect } from "react";
import { useMentionsQuery } from "../../generated/graphql";
import { Box, Grid, useTheme } from "@mui/material";
import InfiniteScroll from "react-infinite-scroll-component";
import { NotificationLite } from "./NotificationLite";
import { headerOffset } from "../../utils/boom-web3/constants";

export const Notifications = () => {
  const theme = useTheme();
  const { loading, data, fetchMore } = useMentionsQuery({
    variables: {
      offset: 0,
      limit: 10,
    },
  });

  console.log(data);

  const fetchData = () => {
    fetchMore({
      variables: {
        offset: data?.mentions?.length ?? 0,
      },
    });
  };

  if (loading) return <Loader />;
  return (
    <Box
      id="notificationScroll"
      maxWidth="25rem"
      sx={{
        p: 2,
        borderLeft: `1px solid ${theme.tertiaryColor}`,
        height: headerOffset,
        overflow: "auto",
      }}
    >
      {data?.mentions?.length ? (
        <InfiniteScroll
          dataLength={data?.mentions?.length}
          next={fetchData}
          hasMore={true}
          scrollableTarget="notificationScroll"
          loader={
            loading && (
              <Box sx={{ marginTop: "1rem" }}>
                <Loader />
              </Box>
            )
          }
        >
          {data.mentions.length
            ? data.mentions.map((mention: any) => (
                <NotificationLite key={mention.id} mention={mention} />
              ))
            : null}
        </InfiniteScroll>
      ) : (
        <CustomResponse text="Follow some people to get some feed updates" />
      )}
    </Box>
  );
};
