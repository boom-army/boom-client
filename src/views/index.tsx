export { BoomOnes } from "./BoomOnes";
export { ChannelFeed } from "./ChannelFeed";
export { Channels } from "./Channels";
export { PeopleView } from "./People";
export { Explore } from "./Explore";
export { Following } from "./Following";
export { Dashboard } from "./Dashboard";
export { News } from "./News";
export { HeroFeed } from "./HeroFeed";
// export { OGMint } from "./OGMint";
