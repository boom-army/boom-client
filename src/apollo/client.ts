import { ApolloClient, InMemoryCache } from "@apollo/client";
import { offsetLimitPagination } from "@apollo/client/utilities";

export const client = new ApolloClient({
  cache: new InMemoryCache({
    typePolicies: {
      Query: {
        fields: {
          feed: offsetLimitPagination(),
          heroFeed: offsetLimitPagination(),
          getChannelById: offsetLimitPagination(),
          users: offsetLimitPagination(),
          searchTweets: offsetLimitPagination(),
          searchUser: offsetLimitPagination(),
          mentions: offsetLimitPagination(),
        },
      },
    },
  }),
  credentials: 'include',
  uri: process.env.REACT_APP_APOLLO_API || "http://locahost:7777",
  headers: {
    authorization: localStorage.getItem("token") || "",
    "Access-Control-Allow-Origin" : "*",
  },
  defaultOptions: {
    watchQuery: {
      context: async ({ req }: any) => {
        const token = req?.headers?.authorization || null;
        const user = token ? localStorage.getItem("user")?.toString() : null;
        return { user, isLoggedIn: !!token };
      },
    },
  },
});
