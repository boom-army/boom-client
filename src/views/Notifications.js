import { useEffect } from "react";
import { CustomResponse } from "../components/CustomResponse";
import { Loader } from "../components/Loader";
import { ShowTweet } from "../components/Tweet";
import styled from "styled-components";
import { MENTIONS } from "../queries/others";
import { useQuery } from "@apollo/client";

const Wrapper = styled.div``;

export const Notifications = ({ refetchProfile }) => {
  const { loading, data } = useQuery(MENTIONS);

  useEffect(() => {
    refetchProfile && refetchProfile();
  }, [data, refetchProfile]);

  if (loading) return <Loader />;

  return (
    <Wrapper>
      {data?.mentions?.length ? (
        data.mentions.map((tweet) => <ShowTweet key={tweet.id} tweet={tweet} />)
      ) : (
        <CustomResponse text="Follow some people to get some feed updates" />
      )}
    </Wrapper>
  );
};
