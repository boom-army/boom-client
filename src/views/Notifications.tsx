import InfiniteScroll from "react-infinite-scroll-component";
import React, { useEffect } from "react";
import { Box, Grid } from "@mui/material";
import { CustomResponse } from "../components/CustomResponse";
import { Loader } from "../components/Loader";
import { Notification } from "../components/Notification/Notification";
import { useMentionsQuery } from "../generated/graphql";

export const Notifications = ({ refetchProfile }: any) => {
  const { loading, data, fetchMore } = useMentionsQuery({
    variables: {
      offset: 0,
      limit: 10,
    },
  });

  useEffect(() => {
    refetchProfile && refetchProfile();
  }, [data, refetchProfile]);

  const fetchData = () => {
    fetchMore({
      variables: {
        offset: data?.mentions?.length ?? 0,
      },
    });
  };

  if (loading) return <Loader />;
  return (
    <Grid item xs={12}>
      {data?.mentions?.length ? (
        <InfiniteScroll
          dataLength={data?.mentions?.length}
          next={fetchData}
          hasMore={true}
          scrollableTarget="scrollBox"
          loader={
            loading && (
              <Box sx={{ marginTop: "1rem" }}>
                <Loader />
              </Box>
            )
          }
        >
          {data.mentions.length
            ? data.mentions.map((mention: any) => (
                <Notification key={mention.id} mention={mention} />
              ))
            : null}
        </InfiniteScroll>
      ) : (
        <CustomResponse text="Follow some people to get some feed updates" />
      )}
    </Grid>
  );
};
