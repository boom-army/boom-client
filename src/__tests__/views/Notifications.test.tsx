import React from "react";
import { Notifications } from "../../views/Notifications";
import renderer from "react-test-renderer";
import { MockedProvider } from "@apollo/client/testing";
import { MentionsDocument } from "../../generated/graphql";

// export const __mocks__: any = [
//   {
//     request: {
//       query: MentionsDocument,
//       variables: {
//         offset: 0,
//         limit: 10,
//       },
//     },
//     result: {
//       data: {
//         mentions: [],
//       },
//       errors: "An error occurred",
//     },
//   },
// ];

// it("renders correctly", async () => {
//   const component = renderer.create(
//     <MockedProvider mocks={__mocks__} addTypename={false}>
//       <Notifications />
//     </MockedProvider>
//   );

//   await new Promise((resolve) => setTimeout(resolve, 0)); // wait for response

//   const tree = component.toJSON();
//   expect(tree).toMatchSnapshot();
//   // expect(tree).toContain('An error occurred');

//   // const tree = renderer.create(
//   //   <Notifications/>
//   // ).toJSON();
//   // expect(tree).toMatchSnapshot();
// });

// it('displays a loading message when fetching', () => {
//   const component = renderer.create(
//     <MockedProvider mocks={__mocks__} addTypename={false}>
//       <Notifications/>
//     </MockedProvider>,
//   );

//   const tree = component.toJSON();
//   expect(tree).toContain('Loading...');
// });

const data:any= {"data":{"mentions":[{"id":"ckzz3yhuu3478401mnf2zgolb2","text":"Hey @sweet-sky-0445 - is this your first meep! congratulations!!\n","tags":[],"isTweetMine":false,"commentsCount":0,"retweetsCount":0,"isRetweet":false,"tipsCount":"0","reactions":[],"files":[],"user":{"id":"cku97qp1a000301mpyoptpo8e","avatar":"https://i1.sndcdn.com/artworks-gUNkngEJ7WTd043w-kpLWPQ-t500x500.jpg","publicAddress":"BhdKDqh8fMxd9L3ewkBTnZtF22BbTpqZNJxUpJP4W51P","handle":"quiet-salad","consumerName":"QS","__typename":"User"},"createdAt":"1645593628134","__typename":"Tweet"}]}}

it('displays a loading message when fetching', () => {
  const component = renderer.create(
    // <MockedProvider mocks={__mocks__} addTypename={false}>
      <Notifications/>
    // </MockedProvider>,
  );

  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});