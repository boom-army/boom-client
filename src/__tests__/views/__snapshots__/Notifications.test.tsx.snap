// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays a loading message when fetching 1`] = `
<div
  className="MuiBox-root css-1l4w6pd"
>
  <span
    className="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-1tww5sk-MuiCircularProgress-root"
    role="progressbar"
    style={
      Object {
        "height": 30,
        "width": 30,
      }
    }
  >
    <svg
      className="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
        cx={44}
        cy={44}
        fill="none"
        r={20.2}
        strokeWidth={3.6}
        style={Object {}}
      />
    </svg>
  </span>
</div>
`;
